sklearn/datasets/_base.py

```diff
@@ -201,6 +209,10 @@ def load_files(
         filenames: ndarray
             The filenames holding the dataset.
     """
+
+    if allowed_extensions is not None:
+        allowed_extensions = frozenset(allowed_extensions)
```

Comment:
To keep with the "style" of this function, let's move this to after the `if categories is not None` check:

https://github.com/scikit-learn/scikit-learn/blob/abbee570f31a91243c22b1892e42056bb915c056/sklearn/datasets/_base.py#L212-L213